internal/model/matcher.go

```diff
@@ -2,6 +2,14 @@ package model
 
 type PathMatcher interface {
 	Matches(f string, isDir bool) (bool, error)
+
+	// Express this PathMatcher as a sequence of filepath.Match
+	// patterns. These patterns are widely useful in Docker-land because
+	// they're suitable in .dockerignore or Dockerfile ADD statements
+	// https://docs.docker.com/engine/reference/builder/#add
+	//
+	// Returns false if this PathMatcher is not expressible this way.
+	AsMatchPatterns() ([]string, bool)
```

Comment:
Should this be a separate interface, since it doesn't apply in all cases?

Reply:
i like that idea, done!